sdk/go.mod

```diff
@@ -1,9 +1,57 @@
 module github.com/pulumi/pulumi-google-native/sdk
 
-go 1.15
+go 1.17
```

Comment:
I suggest we keep this in line with https://www.pulumi.com/docs/intro/languages/go/ - so 1.16+. This will require go programs using this sdk be at 1.17